File: 프로그래머스/lv2/42587. 프린터/프린터.py
from collections import deque

def solution(priorities, location):
    ans = []
    que = deque()
    for i in range(len(priorities)):
        que.append([priorities[i], i])
    
    while que:
        q = que.popleft()
        if que and max(que)[0] > q[0]:
            que.append(q)
        else:
            ans.append(q)

    for i in ans:
        if i[1] == location:
            return ans.index(i) + 1

File: 프로그래머스/lv2/42587. 프린터/프린터.js
function solution(priorities, location) {
  let ans = 0;
  let stack = [];

  priorities.forEach((value, index) => {
    stack.push([value, index]);
  });
  while (stack.length > 0) {
    tmp = stack.shift();
    if (
      stack.length > 0 &&
      Math.max(...stack.map((value) => value[0])) > tmp[0]
    ) {
      stack.push(tmp);
    } else {
      ans++;
      if (tmp[1] == location) {
        return ans;
      }
    }
  }
}